### PR TITLE
site: sync all code type of demos when change one of them

### LINKED ISSUE #48414: 文档demo 的TS/JS 语法切换 支持全局设置.
### What problem does this feature solve?

个人感觉 对demo而言 js模式更适合快速查看/理解代码 因为没有ts的类型代码减少了视野干扰
但目前每个demo的语言模式 是单独控制的  所以每个都要点一遍 比较麻烦 不那么友好

### What does the proposed API look like?

新增一个新的全局切换按钮,
或者
在现在按钮的基础上 默认全局切换 并缓存本地

### PR DESCRIPTION
**Pull request by gin-lsl**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [x] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link
切换 Demo 的 TypeScript / JavaScript 时，可以全局切换所有的。

close #48414 

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |           |
| 🇨🇳 Chinese |           |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

File: .dumi/theme/slots/Content/index.tsx [hunk 1]
```diff
@@ -55,10 +60,14 @@ const Content: React.FC<React.PropsWithChildren> = ({ children }) => {
...
 
   const contextValue = useMemo<DemoContextProps>(
-    () => ({ showDebug, setShowDebug }),
-    [showDebug, debugDemos],
+    () => ({ showDebug, setShowDebug, codeType, setCodeType }),
+    [showDebug, codeType, debugDemos],
   );
 
+  useEffect(() => {
+    localStorage.setItem(DEFAULT_DEMOS_CODE_TYPE, codeType);
```
Comment (afc163) [inline .dumi/theme/slots/Content/index.tsx:68]: 别用 localStorage，放到 location query 上去。

Comment (afc163) [reply .dumi/theme/slots/Content/index.tsx:68]: 或者感觉没必要记这个状态。

**Comment by afc163** [comment]
图片

https://github.com/ant-design/ant-design/actions/runs/8673975635/job/23785853683?pr=48436

File: .dumi/theme/slots/Content/index.tsx (hunk 1 above)
Comment (gin-lsl) [reply .dumi/theme/slots/Content/index.tsx:68]: 有道理，我删掉 localStorage 吧

**Review by PeachScript — APPROVED** [review]

**Comment by li-jia-nan** [comment]
这个交互有点反直觉啊……太奇怪了，我还以为是单独加个总按钮什么的

**Comment by yoyo837** [comment]
> 这个交互有点反直觉啊……太奇怪了，我还以为是单独加个总按钮什么的

site的，影响面小，反对的声音多了再改 😂

**Pull request merged**